bindings/cpp/.clang-format

```diff
@@ -0,0 +1,3 @@
+---
```

Comment:
c binding is using `Webkit` style:

https://github.com/apache/incubator-opendal/blob/main/bindings/c/Makefile#L34-L37

Maybe we can align with c binding?

Reply:
CMake not contains a format job. Webkit is a niche coding style, more projects use Google coding style.

Reply:
> CMake not contains a format job.

We can add a custom target in cmake to let user able to use `make format` directly. e.x.
```
add_custom_target(
    format
    ALL
    COMMAND /usr/bin/clang-format
    -i
    ${CPP_SOURCE_FILE}
    ${CPP_HEADER_FILE}
)
```

Also, Would you like to update ci to check format?

Reply:
`COMMAND /usr/bin/clang-format` seems not cool as not all users install `clang-format` at this location.

Reply:
> Also, Would you like to update ci to check format?

Please don't extend contributor's work in PR review. We can start a new issue for this.

Reply:
> COMMAND /usr/bin/clang-format seems not cool as not all users install clang-format at this location.

We can replace it with `clang-format`. cmake will try finding it in $PATH

Reply:
LGTM

Reply:
cc @JackDrogon what do you think?

Reply:
It's not a good choice, format code can be handle in git pre-commit.

Reply:
We need to add a format code ci pipeline.